packages/tpetra/core/src/Tpetra_Details_allReduceView.hpp

```diff
@@ -182,17 +182,11 @@ allReduceView (const OutputViewType& output,
     return;
   }
 
-  // We've had some experience that some MPI implementations do not
-  // perform well with UVM allocations.
-  const bool assumeMpiCanAccessBuffers =
-    ! view_is_cuda_uvm<OutputViewType>::value &&
-    ! view_is_cuda_uvm<InputViewType>::value &&
-    ::Tpetra::Details::Behavior::assumeMpiIsCudaAware ();
```

Comment:
What happens if we leave in `assumeMpiIsCudaAware` and take out the "not UVM" requirement?  Does it still work?  I'd rather not break the case where MPI really can't access CUDA buffers.

Reply:
See final commit